### PR TITLE
refactor: Application Group Overview Table Revamp, feat: create EnvironmentOverviewTable

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ApplicationGroup/Details/EnvironmentOverview/envOverview.scss [hunk 1]
```diff
@@ -16,34 +16,11 @@
...
-
-            .app-deployment-info-row-leftsection {
-                grid-template-columns: 10% 10% 80%;
-                column-gap: 16px;
-                min-width: 170px;
-            }
-        }
-    }
+    flex-grow: 1;
+    overflow: auto;
```
Comment (eshankvaish) [inline src/components/ApplicationGroup/Details/EnvironmentOverview/envOverview.scss:23]: - Check for using from utils
- Check if any of the css can be removed

File: src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx [hunk 2]
```diff
@@ -206,52 +189,34 @@ export default function EnvironmentOverview({
...
-    const sortByDeployedAt = () => {
-        handleSorting(EnvironmentOverviewSortableKeys.deployedAt)
-    }
+    const getAppRedirectLink = (appId: number, envId: number) => `/app/${appId}/details/${envId}`
 
     const sortAndUpdateAppListData = (_appListData) => {
+        const { sortBy = EnvironmentOverviewTableSortableKeys.NAME, sortOrder = SortingOrder.ASC } = searchParams
         setAppListData({
             ..._appListData,
-            appInfoList: _appListData.appInfoList.sort((a, b) => {
```
Comment (eshankvaish) [inline src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx:253]:
```suggestion
            appInfoList: _appListData.appInfoList.sort((a, b) => {
```

File: src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx [hunk 3]
```diff
@@ -266,7 +231,7 @@ export default function EnvironmentOverview({
         if (appListData) {
             sortAndUpdateAppListData(appListData)
         }
-    }, [sortBy, sortOrder])
+    }, [searchParams.sortBy, searchParams.sortOrder])
```
Comment (eshankvaish) [inline src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx:234]: We can possibly move to the component; and use memo

File: src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx [hunk 4]
```diff
@@ -336,87 +301,32 @@ export default function EnvironmentOverview({
...
-        )
+    const openCommitInfoModal = (ciArtifactId: number) => (e) => {
+        e.stopPropagation()
+        setCommitInfoModalConfig({
+            envId,
+            ciArtifactId: ciArtifactId,
+        })
     }
 
+    const environmentOverviewTableRows = appListData?.appInfoList?.map((appInfo) => ({
```
Comment (eshankvaish) [inline src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx:312]: add type

File: src/Pages/Shared/EnvironmentOverviewTable/EnvironmentOverviewTable.component.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+                                            {Icon && (
+                                                <Icon
+                                                    className={`icon-dim-16 ${iconType === 'fill' ? 'fcn-7' : ''} ${iconType === 'stroke' ? 'scn-7' : ''}`}
+                                                />
+                                            )}
+                                            <span className="dc__truncate cn-9 fs-13 lh-20">{label}</span>
+                                        </button>
+                                    ))}
+                                </PopupMenu.Body>
+                            </PopupMenu>
```
Comment (eshankvaish) [inline src/Pages/Shared/EnvironmentOverviewTable/EnvironmentOverviewTable.component.tsx:196]: suggestion: abstract to a separate component

**Review by eshankvaish — CHANGES_REQUESTED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**